handbook/openid-connect/openid-connect.md

```diff
@@ -0,0 +1,133 @@
+- Administrator desktop access to the SystemLink server
+- A OpenID Connect Provider server such as [PingFederate](https://www.pingidentity.com/en/software/pingfederate.html), [Azure ADFS](https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/deployment/how-to-connect-fed-azure-adfs), [Okta](https://www.okta.com/openid-connect/), or another [certified provider](https://openid.net/certification/) that has been fully setup and configured for OpenID Connect authentication. 
+    - If you have not yet setup your provider please consult the vendor documentation for setup and configuration. 
+
+## Enabling OpenID Connect in SystemLink
+1. Log into the server running SystemLink and open **NI Web Server Configuration**
+2. Go to the **Authentication** tab and enable **Use OpenID Connect (advanced)**
+
+![Enable OpenID Connect in NI Web Server](odic-webserver.png)
+*Enable OpenID Connect in NI Web Server*
+
+## OpenID Configuration Files in SystemLink server
+
+There are three files that must be created to connect your SystemLink server to an OpenID Connect provider: `[provider-dns].conf`, `[provider-dns].client`, and `[provider-dns].provider`. The `[provider-dns]` portion of each filename must be the URL encoded fully qualified domain name. 
+
+> For example, a OpenID provider with the DNS `openid-provider.com` would yield files named `openid-provider.com/conf` , `openid-provider.com.client`, and `openid-provider.com.provider`. 
+
+These files do no exist for new SystemLink installations. Each file must be added to the `C:\Program Files\National Instruments\Shared\Web Server\conf\openidc` directory. The NI Web Server must be restarted for these changes to take effect.
+
+SystemLink can be configured to support multiple OpenID Providers simultaneously by creating a `[provider-dns].conf`, `[provider-dns].client`, and `[provider-dns].provider` file for each provider.  It is required that the user ID used within SystemLink be unique across providers. This id take the form `[sub_claim]@issuer` where this issuer is the DNS name of the OpenID Connect provider. You can see the id SystemLink uses for a user by viewing the details of that user in the SystemLink security application. 
+
+### SystemLink Login Window Configuration
+
+`[provider-dns].conf` describes the scopes SystemLink will request, and the text and icon for the provider login button. 
+```json
+{
+ "scope": "openid email profile",
+ "ni-attributes": {
+   "displayName": "Log in with PingFederate",
+   "iconUri": "/login/assets/pf.png"
+ }
+}
+```
+In this example the `openid`, `email`, and `profile` scopes are requested. Additional scopes may be requested. Consult your provider's documentation regarding exposing scopes to clients. Each scope will contain claims that can be used to map to roles within workspaces in SystemLink. See [**Mapping OpenID Connect Claims to SystemLink Workspaces and Roles**](#mapping-openid-connect-claims-to-systemlink-workspaces-and-roles) for details. 
```

Comment:
Might be worth mentioning that the `profile` and `email` scopes provide the claims SystemLink uses for the user's first/last name and email.

Reply:
Good idea. Done!